### PR TITLE
Api tidyup

### PR DESCRIPTION
**Pull request by xen-0**

### CODE REVIEW

File: scanpointgenerator/core/compoundgenerator.py [hunk 1]
```diff
@@ -16,48 +16,57 @@ class CompoundGenerator(object):
 
     typeid = "scanpointgenerator:generator/CompoundGenerator:1.0"
 
-    def __init__(self, generators, excluders, mutators):
+    def __init__(self, generators, excluders, mutators, duration=0.05):
```
Comment (thomascobb) [inline scanpointgenerator/core/compoundgenerator.py]: Default should be -1

File: scanpointgenerator/core/dimension.py [hunk 2]
```diff
@@ -1,13 +1,64 @@
...
+        """
+        Retrieve the positions for a given axis within the dimension.
+
+        Args:
+            axis (str): axis to get positions for
+        Returns:
+            Positions (np.array): Array of positions
+        """
+        # the points for this axis must be scaled and then indexed
+        if self._full_mask is None: self.create_dimension_mask()
```
Comment (thomascobb) [inline scanpointgenerator/core/dimension.py]: Can we fail here instead please. Anything apart from prepare*() and __init__() should be threadsafe. Can we turn create_dimension_mask() into prepare_dimension_mask() to be consistent please?

File: scanpointgenerator/generators/lissajousgenerator.py [hunk 3]
```diff
@@ -9,27 +9,27 @@
...
-            num_points=None, alternate_direction=False):
+    def __init__(self, axes, units, box, num_lobes,
+            size=None, alternate=False):
         """
         Args:
-            names (list(str)): The scannable names e.g. ["x", "y"]
-            units (str): The scannable units e.g. "mm"
+            axes (list(str)): The scannable axes e.g. ["x", "y"]
+            units (list(str)): The scannable units e.g. ["mm", "mm"]
             box(dict): Dictionary of centre, width and height representing
```
Comment (thomascobb) [inline scanpointgenerator/generators/lissajousgenerator.py]: box dict now looks inconsistent with units, axes, etc. Can we have centre and span 2 element lists instead please?

File: scanpointgenerator/generators/lissajousgenerator.py [hunk 4]
```diff
@@ -9,27 +9,27 @@
...
                 box to fill with points
             num_lobes(int): Number of x-direction lobes for curve; will
```
Comment (thomascobb) [inline scanpointgenerator/generators/lissajousgenerator.py]: And can we turn num_lobes into lobes please?

File: scanpointgenerator/generators/lissajousgenerator.py (hunk 3 above)
Comment (xen-0) [inline scanpointgenerator/generators/lissajousgenerator.py]: Wouldn't top_right and bottom_left lists be easier to use? Defining the opposite corners is certainly easier.

Comment (thomascobb) [inline scanpointgenerator/generators/lissajousgenerator.py]: That would work

Comment (xen-0) [inline scanpointgenerator/generators/lissajousgenerator.py]: I thought this was the bounding box for Dimension, not the LissajousGenerator.
In which case, I'm not sure which is conceptually more convenient, but I'd be inclined to use the centre/span definition for now.

File: scanpointgenerator/generators/lissajousgenerator.py [hunk 5]
```diff
@@ -38,24 +38,21 @@ def __init__(self, names, units, box, num_lobes,
...
             self.size = num_lobes * 250
         self.increment = 2*m.pi/self.size
 
-        self.position_units = {self.names[0]: units, self.names[1]: units}
         self.index_dims = [self.size]
         gen_name = "Lissajous"
-        for axis_name in self.names[::-1]:
+        for axis_name in self.axes[::-1]:
             gen_name = axis_name + "_" + gen_name
         self.index_names = [gen_name]
```
Comment (thomascobb) [inline scanpointgenerator/generators/lissajousgenerator.py]: Index names can go everywhere...

File: scanpointgenerator/generators/lissajousgenerator.py [hunk 6]
```diff
@@ -38,24 +38,21 @@ def __init__(self, names, units, box, num_lobes,
...
         # Phase needs to be 0 for even lobes and pi/2 for odd lobes to start
         # at centre for odd and at right edge for even
         self.phase_diff = m.pi/2 * (num_lobes % 2)
-        if num_points is None:
+        if size is None:
             self.size = num_lobes * 250
         self.increment = 2*m.pi/self.size
 
-        self.position_units = {self.names[0]: units, self.names[1]: units}
         self.index_dims = [self.size]
```
Comment (thomascobb) [inline scanpointgenerator/generators/lissajousgenerator.py]: and index_dims too

**Comment by thomascobb** [comment]
Malcolm pull request is here in case you're interested...

https://github.com/dls-controls/pymalcolm/pull/202

Wehn you have done this and released scanpointgenerator as 2.0 then the I can merge Malcolm's pull request...

**Pull request merged**